### DISCUSSION TITLE
Poly Hulls in Sub Editor

### DISCUSSION THREAD

**Discussion by JamesTCrenshaw** [FakeFishGames/Barotrauma#6172] [Ideas and suggestions]
Instead of perfectly square Hulls in the sub editor, perhaps they could be polygons? I am unsure if this is even possible to be implemented.
Edit: To clarify, I do mean the "Hull" box that denotes separate rooms.

**Comment by ViktorShield** [comment]
Yeah, I believe that would be a real pain in the ass to be implemented. But as of now, any shape complex enough becomes an annoyance to build. Also, light behaves strangely in that case

**Comment by MaaiG** [comment]
If a polygon cannot be made, an `additional Hull` as a triangle may solve the problem. Would a right triangle have a problem with light? Because that’s exactly half a rectangle.
